kclvm/cmd/src/tests.rs

```diff
@@ -97,3 +99,73 @@ fn settings_arguments(path: std::path::PathBuf) -> Vec<String> {
         "a.b.c".to_string(),
     ]
 }
+
+#[test]
+fn test_external_cmd() {
+    let matches = app().get_matches_from(&[ROOT_CMD, "run", "-E", "test_name=test_path"]);
+    let matches = matches.subcommand_matches("run").unwrap();
+    let pair = hashmaps_from_matches(matches, "package_map")
+        .unwrap()
+        .unwrap();
+    assert_eq!("{\"test_name\": \"test_path\"}", format!("{:?}", pair));
+}
+
+#[test]
+fn test_multi_external_cmd() {
+    let matches = app().get_matches_from(&[
+        ROOT_CMD,
+        "run",
+        "-E",
+        "test_name=test_path",
+        "-E",
+        "test_name1=test_path1",
+    ]);
+    let matches = matches.subcommand_matches("run").unwrap();
+    let pair = hashmaps_from_matches(matches, "package_map")
+        .unwrap()
+        .unwrap();
+    assert_eq!(
+        "{\"test_name1\": \"test_path1\", \"test_name\": \"test_path\"}",
+        format!("{:?}", pair)
+    );
+}
+
+#[test]
+fn test_multi_external_with_same_key_cmd() {
+    let matches = app().get_matches_from(&[
+        ROOT_CMD,
+        "run",
+        "-E",
+        "test_name=test_path",
+        "-E",
+        "test_name=test_path1",
+    ]);
+    let matches = matches.subcommand_matches("run").unwrap();
+    let pair = hashmaps_from_matches(matches, "package_map")
+        .unwrap()
+        .unwrap();
+    assert_eq!("{\"test_name\": \"test_path1\"}", format!("{:?}", pair));
```

Comment:
Do not directly compare format strings, as HashMap does not guarantee the order of keys.